src/AutoRest.CSharp/Common/Generation/Writers/CodeWriterExtensions.cs

```diff
@@ -269,7 +273,7 @@ public static void WriteVariableAssignmentWithNullCheck(this CodeWriter writer,
                 }
 
                 var defaultValue = parameter.DefaultValue.Value;
-                if (defaultValue.IsNewInstanceSentinel || TypeFactory.IsExtendableEnum(parameter.Type) || parameter.DefaultValue.Value.Type.Equals(parameter.Type))
+                if (defaultValue.IsNewInstanceSentinel || TypeFactory.IsExtendableEnum(parameter.Type) || parameter.DefaultValue.Value.Type.Equals(parameter.Type) || parameter.Type.Equals(typeof(string)))
```

Comment:
Why `parameter.DefaultValue.Value.Type.Equals(parameter.Type)` didn't work here?

Reply:
(ignoreNullable || IsNullable == other.IsNullable)

param.DefaultValue.Value.Type.IsNullable == false
parameter.Type.IsNullable == true

therefore equals fails

Reply:
Looks like a bug to me. If parameter is nullable but default value is not, we should still be able to assign it. Maybe something like:
```cs
if (defaultValue.IsNewInstanceSentinel || TypeFactory.IsExtendableEnum(parameter.Type) || defaultValue.Type.Equals(parameter.Type) || defaultValue.Type.Equals(parameter.Type.WithNullable(false))
```